catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/StorableResource.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.metacard;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.apache.commons.io.IOUtils;
+import org.codice.ddf.platform.util.TemporaryFileBackedOutputStream;
+
+/** Call {@link #close()} if {@link #isError()} returns {@code false}. */
+public class StorableResource implements AutoCloseable {
```

Comment:
❗️ There should be a `StorableResource` interface since it is used by the `Splitter` interface.

Reply:
done